gamestonk_terminal/cryptocurrency/coinpaprika/coinpaprika_model.py

```diff
@@ -0,0 +1,667 @@
+from datetime import datetime, timedelta
+import textwrap
+import requests
+import pandas as pd
+from requests.adapters import HTTPAdapter
+
+ENDPOINTS = {
+    "global": "/global",
+    "coin": "/coins/{}",
+    "coins": "/coins",
+    "coin_tweeter": "/coins/{}/twitter",
+    "coin_events": "/coins/{}/events",
+    "coin_exchanges": "/coins/{}/exchanges",
+    "coin_markets": "/coins/{}/markets",
+    "ohlcv": "/coins/{}/ohlcv/latest",
+    "ohlcv_hist": "/coins/{}/ohlcv/historical",
+    "people": "/people/{}",
+    "tickers": "/tickers",
+    "ticker_info": "/tickers/{}",
+    "exchanges": "/exchanges",
+    "exchange_info": "/exchanges/{}",
+    "exchange_markets": "/exchanges/{}/markets",
+    "contract_platforms": "/contracts",
+    "contract_platform_addresses": "/contracts/{}",
+    "search": "/search",
+}
+
+PAPRIKA_BASE_URL = "https://api.coinpaprika.com/v1"
+
+# Mount a session with adapter. It's solution for time to time timeouts
+session = requests.Session()
+session.mount(PAPRIKA_BASE_URL, HTTPAdapter(max_retries=5))
+
+
+def make_request(endpoint, payload=None, **kwargs):
+    """Helper method that handles request for coinpaprika api.
+    It prepares URL for given endpoint and payload if it's part of requests
+
+    Parameters
+    ----------
+    endpoint: str,
+        it's an endpoint that we want to query. e.g. to get twitter data for given coin we need to use:
+       https://api.coinpaprika.com/v1/coins/{}/twitter
+    payload: dict
+        the body of your request. Contains the data send to the CoinPaprika API when making an API request
+    kwargs:
+        additional parameters that will be added to payload
+    Returns
+    -------
+    dict with response data
+
+    """
+    url = f"{PAPRIKA_BASE_URL}{endpoint}"
+    if payload is None:
+        payload = {}
+    if kwargs:
+        payload.update(kwargs)
+    return session.get(url, params=payload).json()
+
+
+def get_global_market():
+    """Return data frame with most important global crypto statistics like:
+    market_cap_usd, volume_24h_usd, bitcoin_dominance_percentage, cryptocurrencies_number,
+    market_cap_ath_value, market_cap_ath_date, volume_24h_ath_value, volume_24h_ath_date,
+    market_cap_change_24h, volume_24h_change_24h, last_updated,
+
+    Returns
+    -------
+    pandas.DataFrame
+        Metric, Value
+    """
+    global_markets = make_request(ENDPOINTS["global"])
+    global_markets["last_updated"] = datetime.fromtimestamp(
+        global_markets["last_updated"]
+    )
+    df = pd.Series(global_markets).to_frame().reset_index()
+    df.columns = ["Metric", "Value"]
```

Comment:
Do you think we could change date-time from `2021-05-02T17:00:00Z` to `2021-05-02 17:00:00`? It's more visible